### PR TITLE
test: Fix several integration tests for SparkOfflineStore plugin

### PR DESCRIPTION
**Pull request by jbvaningen**
Ran into some git issues on previous branch from my fork, moved changes here from https://github.com/feast-dev/feast/pull/2460

Signed-off-by: Joost van Ingen <joostingen@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

**Review by achals — COMMENTED** [review]
I took a cursory look and it looks fine to me; but I'll let @kevjumba take a more detailed review

**Comment by achals** [comment]
Also, @jbvaningen the linter is failing and probably needs to be fixed.

File: protos/feast/core/DataSource.proto [hunk 1]
```diff
@@ -169,6 +170,21 @@ message DataSource {
     string warehouse = 5;
   }
 
+  // Defines options for DataSource that sources features from a Snowflake Query
```
Comment (kevjumba) [inline protos/feast/core/DataSource.proto:173]: nit: Spark Query

File: sdk/python/feast/infra/offline_stores/contrib/spark_offline_store/spark_source.py [hunk 2]
```diff
@@ -75,26 +74,11 @@ def __init__(
             "This API is unstable and it could and most probably will be changed in the future.",
             RuntimeWarning,
         )
-        self.allowed_formats = [format.value for format in SparkSourceFormat]
```
Comment (kevjumba) [inline sdk/python/feast/infra/offline_stores/contrib/spark_offline_store/spark_source.py:78]: Why remove these valueerror checks?

Comment (kevjumba) [reply sdk/python/feast/infra/offline_stores/contrib/spark_offline_store/spark_source.py:78]: Oh I see, these are moved to SparkOptions.

**Review by kevjumba — COMMENTED** [review]
1 small nit, but overall looks good, also could you please install the custom pre commit hook that we use, most of the changes here are formatting changes that we don't want to merge into master since that will create large rebase conflicts with other branches @jbvaningen 
Just install the precommit hook and then run `make format `

**Comment by jbvaningen** [comment]
@kevjumba thanks for you patience with the linting errors, I'm still finding my way around the codebase. I rebased on master, resolving the conflicts, and am pretty sure that the issues should be addressed now!

**Comment by kevjumba** [comment]
@jbvaningen Hi could you run `make format` one more time? There seems to be some other lint errors.

**Comment by jbvaningen** [comment]
Weird I just merged the latest upstream changes and now the linting seems ok 🤔

**Comment by kevjumba** [comment]
@jbvaningen Sorry master keeps moving but there's a merge conflict.

**Comment by jbvaningen** [comment]
I'm really not sure what is going on with all the conflicts, but the git state of this branch is really messy again. I have once again moved changes to another PR to keep the history clean 

See https://github.com/feast-dev/feast/pull/2516

**Pull request closed without merging**